resfit/fit_resonator/fit_S_data.py

```diff
@@ -777,7 +777,44 @@ def monte_carlo_fit(xdata= None,ydata=None,parameter=None,Method = None):
         stop_MC = True
     return parameter,stop_MC, error
 
-
+def get_header(line: str):
+    if (line.strip()[0:1]!='#'):
+        print("Header not found in file.")
+        quit()
+    line = line[2:]
+    if line[0:2].lower()=='hz':
+        frequency_units=line[0:2].lower()
+        line = line[3:]
+    elif line[0:3].lower()=='khz':
```

Comment:
You could cover all these cases by:

```python
next_line = line[0:3].lower()
if next in ['khz','ghz','hz']
    frequency_units = next_line
    line = line[4:]
else:
    raise ValueError('Frequency units not found')
```

Reply:
The reason I had them separated out was because hz is only 2 characters long while the others are 3.